client/ayon_houdini/plugins/publish/collect_houdini_batch_families.py

```diff
@@ -0,0 +1,19 @@
+import pyblish.api
+
+
+class CollectNoProductTypeFamilyGeneric(pyblish.api.InstancePlugin):
+    """Collect data for caching to Deadline."""
+
+    order = pyblish.api.CollectorOrder - 0.49
+    families = ["generic"]
+    hosts = ["houdini"]
+    targets = ["local", "remote"]
+    label = "Collect Data for Cache"
+
+    def process(self, instance):
+        # Do not allow `productType` to creep into the pyblish families
+        # so that e.g. any regular plug-ins for `pointcache` or alike do
+        # not trigger.
+        instance.data["family"] = "generic"
+        # TODO: Do not add the dynamic 'rop' family in the collector?
+        instance.data["families"] = ["generic", "rop"]
```

Comment:
Actually - I didn't do that, because I believe that'd make it so that also the instance anatomy data collected for the instance would be the "generic" product type (which may affect the names for your resulting products or output files (I'm not entirely sure).

I recall there being a reason for doing it 'later' in publishing - but I'll try to reproduce some issues and maybe add a comment. But if it does work, and there are no side effects (🥁) then I'd love to remove this and rely on the Creator, yes.

Reply:
I had a look at this - unfortunately we can't currently.

[This is the Collector](https://github.com/ynput/ayon-core/blob/2072f2fbe2bdaa09c96dd3c849bff65785c3b3bb/client/ayon_core/plugins/publish/collect_from_create_context.py#L89-L91) that turns the `CreateContext` 's `CreatedInstance` items into the pyblish instances. As you can see it sets the instance's family to the product type - which makes sense, but also means that we can't override the family explicitly separate from the product type.

We want to preserve e.g. product type "pointcache" because that's what the publisher attributes are also based on (like certain export options specific for a product type) whilst avoiding the usage of the regular product type plug-ins. We may need to redesign how the familie work for the 'original legacy rop instances' so that we maybe can differentiate which plug-ins are exposed where.

This is again, a CreateContext / Publisher system design issue or I'm missing how I could best redesign this to work around that.

Reply:
Update the code with`TODO` explaining the issue. Whether there are better ways, not sure. @iLLiCiTiT @antirotor